message.go

```diff
@@ -120,6 +121,17 @@ func NewMessage(data []byte) *Message {
 	}
 }
 
+// Null is an AMQP null.
+// Typically used in [Message.Value] to send a null.
+//
+// https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#type-null
+type Null struct{}
```

Comment:
The spec in AMQP defines a `null` type. Is it needed though?. I'm wondering if it makes sense to have that type verse having the developer write
```
err = sender.Send(ctx, &amqp.Message{Value: nil}, nil)
```

Reply:
Unfortunately, it's ambiguous as `Value` is defined as `any` so `nil` is its zero-value.

In `Message` there are fields `Data`, `Value`, and `Sequence` and they're all mutually exclusive. So, we must explicitly set one of them when creating the message and the others are left as `nil` meaning "skip this field".